tests/components/sunweg/test_init.py

```diff
@@ -76,6 +76,22 @@ async def test_sunwegdata_update_success(plant_fixture) -> None:
     assert len(data.data.inverters) == 1
 
 
+async def test_sunwegdata_update_success_alternative(plant_fixture_alternative) -> None:
+    """Test SunWEGData success on update."""
+    api = MagicMock()
+    api.plant = MagicMock(return_value=plant_fixture_alternative)
+    api.complete_inverter = MagicMock()
+    data = SunWEGData(api, 0)
```

Comment:
Please don't interact with integration details directly in the tests. Set up the integration and assert state in the state machine instead.